src/figuresClasses.ts

```diff
@@ -1,11 +1,104 @@
-export interface Figure {}
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('sides must be greater than zero');
+    }
+
+    if (c >= a + b || a >= b + c || b >= a + c) {
+      throw new Error('wrong length');
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const halfP = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      halfP * (halfP - this.a) * (halfP - this.b) * (halfP - this.c),
+    );
+
+    return parseFloat(area.toFixed(2));
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  shape: 'circle';
 
-export class Circle implements Figure {}
+  color: Color;
+
+  radius: number;
+
+  constructor(color: Color, radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than zero');
+    }
+
+    this.shape = 'circle';
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    const area: number = Math.PI * Math.pow(this.radius, 2);
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: 'rectangle';
+
+  color: Color;
+
+  width: number;
+
+  height: number;
+
+  constructor(color: Color, width: number, height: number) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Width and height must be greater than zero');
+    }
+    this.shape = 'rectangle';
+    this.color = color;
+    this.width = width;
+    this.height = height;
+  }
+
+  getArea(): number {
+    const area: number = this.width * this.height;
+
+    return parseFloat(area.toFixed(2));
+  }
+}
 
-export class Rectangle implements Figure {}
+export function getInfo(figure: Figure): string {
+  const area = figure.getArea();
 
-export function getInfo(figure): string {
-  return typeof figure;
+  return `A ${figure.color} ${figure.shape} - ${area}`;
```

Comment:
Consider using template literals for better readability: `return `A ${figure.color} ${figure.shape} - ${area}`;`